src/hazelcore/HTyp.re

```diff
@@ -22,10 +22,10 @@ let is_Prod =
   | Prod(_) => true
   | _ => false;
 
-let precedence_Prod = 1;
```

Comment:
@dmoon1221 Sorry I misswrote, but the precedence was originally (in the dev branch) `Prod < Arrow < Sum < const`. I also wrote a test using the original typ parser as the source of truth. That test passes against my parser only when the type precedences are `Prod < Arrow < Sum`. 

Does this mean that the original parser was just broken? If so I will go ahead and set `const < Prod < Sum < Arrow`, flip the comparisons in `contract`, and re write the test accordingly.

Reply:
Yes, I think it was previously broken. @cyrus-, can you confirm (cf thread since Cory's last commit).

Reply:
Ah. Thank you so much for bearing with me, I thought I was going crazy.

Reply:
Revision to my recommendation after talking to Cyrus: We should still standardize around the `const` precedence having the lowest numeric value, but the previous order where `Arrow` binds more tightly than `Sum` should be preserved (doesn't really matter once we labeled sums, as we'll likely get rid of the `Sum` operator, but we'll preserve existing behavior for now). So the final order is `const < Prod < Arrow < Sum`. Sorry for the confusion.

Reply:
Will do, but how should I adjust comparisons in `UHTyp.contract`? Since the precedence isn't reversed, I can't just flip the comparisons in that function. Should I just fully rewrite contract?

Reply:
Ah, taking a closer look at your diffs, I see now how we encountered this surprising skew in precedence values. `HTyp` used the standard ordering convention for precedence, where a higher value means 'binds tighter', whereas we used the reverse ordering everywhere else (for ease of extensibility). When you transferred `HTyp.precedence_const` to `Operators_Typ.precedence_const` in this [commit](https://github.com/hazelgrove/hazel/pull/361/commits/4617760f6d418a4feaac0f3da5e6650cff75fbf2#diff-21dd9ddc05d40c65ec2e5cb4aaea516fR25), the reversal didn't take place while you had already applied the reversal for the rest of the operators. So changing `precedence_const` to `0` simply completes your intended reversal, and now the operators in `UHTyp.contract` (which were originally based on the `HTyp` standard ordering) just need to be flipped.